### PR TITLE
Switch to new ModDevGradle plugin

### PR DESCRIPTION
**Pull request by Technici4n**
Note that MDG supports parchment, if you want to use it in all projects (especially in common). See https://github.com/neoforged/ModDevGradle?tab=readme-ov-file#better-minecraft-parameter-names--javadoc-parchment.

### CODE REVIEW

File: NeoForgeApi/build.gradle.kts [hunk 1]
```diff
@@ -47,22 +47,14 @@ java {
...
 
-minecraft {
-	accessTransformers {
-		// All minecraft configurations in the multi-project must be identical, including ATs,
-		// because of a ForgeGradle bug https://github.com/MinecraftForge/ForgeGradle/issues/844
-		file("../Forge/src/main/resources/META-INF/accesstransformer.cfg")
-	}
+neoForge {
+	version = neoforgeVersion
+	accessTransformers.from(file("../NeoForge/src/main/resources/META-INF/accesstransformer.cfg"))
```
Comment (mezz) [inline NeoForgeApi/build.gradle.kts]: NeoForgeApi doesn't actually need ATs, this was just included as a workaround for https://github.com/MinecraftForge/ForgeGradle/issues/844

Is this still needed with MDG?

File: build.gradle.kts [hunk 2]
```diff
@@ -4,9 +4,7 @@ plugins {
     // https://maven.fabricmc.net/fabric-loom/fabric-loom.gradle.plugin/
     id("fabric-loom") version("1.7.3") apply(false)
     // https://projects.neoforged.net/neoforged/neogradle
```
Comment (mezz) [inline build.gradle.kts]:
```suggestion
    // https://projects.neoforged.net/neoforged/moddevgradle
```
these URLs are intended to point to the place to find the latest version numbers (like `2.0.26-beta` here)

**Review by mezz — CHANGES_REQUESTED** [review]
Thanks for trying it out. I'm surprised it worked easily.

Were you able to test that the output jars are the same?

Having parchment would be nice. I had it set up before but ran into issues.

File: NeoForgeApi/build.gradle.kts (hunk 1 above)
Comment (Technici4n) [reply NeoForgeApi/build.gradle.kts]: It's not needed, no. I left it in to save one recompilation, but removing it is a bit cleaner. (Once it's in the cache it doesn't matter for subsequent runs.)

**Comment by Technici4n** [comment]
> Were you able to test that the output jars are the same?

Yes, they are the same when published to maven local at least!

File: NeoForgeApi/build.gradle.kts (hunk 1 above)
Comment (mezz) [reply NeoForgeApi/build.gradle.kts]: We can leave it in, that sounds good. Can you add a comment above this line to explain why it's here?

**Review by mezz — APPROVED** [review]

**Pull request merged**